### ISSUE TITLE
Obscure API and very slow dev

### ISSUE THREAD

**Issue by ibc** [MixinNetwork/kraken#4]
https://vec.io/posts/kraken-high-performance-webrtc-sfu

This is a very disrespectful post. It's ok that you discard SFUs just because they are not written in Go (the only language valid for you) but at least do not write shit about them.

A "SFU just for audio" can be done in 15 minutes, so you have done nothing special.

And BTW:

- Unified-Plan is not a "super recent and cool WebRTC feature". It's been there for long years.
- Adding video is not "an easy extension". Well, you'll see when you try to implement it.

In short: you should not write shit about others just because you want to promote your own solution on Go.

**Comment by cedricfung** [comment]
Sorry if I offend you at any words of that post.

I understand for someone like you, an core engineer of the famous mediasoup, you are able to make an audio only SFU from scratch in under 15 minutes, good job. But I must say that I am clearly not smart enough, even dumb to learn mediasoup and get some demo project running up without using your client library. I know you are smartest to make that excellent work in minutes or hours so that nobody could use it in 15 minutes or days.

Of course Kraken may be slow in production, but quick to develop and get shit fixed.

For the unified plan thing, yes it was introduced years ago, and I assume you should know that it's only stable and default in Chrome for no more that 2 years yet. Of course not cutting edge 😂 Janus doesn't support it though, many engineers not smart as you don't even know it.

I write shit, I don't know Kraken is slower than mediasoup, and I don't believe any beginners would think Kraken API is obsecure than any other SFUs, including your mediasoup. 

At least could you please make an working example like the one in Kraken readme, in 15 minutes, without your awesome but too complicated client library?

**Comment by ibc** [comment]
> At least could you please make an working example like the one in Kraken readme, in 15 minutes, without your awesome but too complicated client library?

No, I won't waste my time with people like you. Sorry, nothing personal. It's just that you suck. Bye.

**Comment by cedricfung** [comment]
I'm glad your problem fixed, now I will close  the issue if you don't have any other questions.

**Comment by lminiero** [comment]
"Janus doesn't support it"? Do you even do your homework? We've had a branch ready and under test for weeks. I agree with Iñaki that the blog post was highly disrespectful of the work developers like us have been doing FOR YEARS and given away to thousands of people FOR FREE. I won't even deign the "Janus developers are slow" comment the dignity of a response.

**Comment by cedricfung** [comment]
As I written in the blog post, "We used Janus in our previous WebRTC application years ago, and it’s our first candidate when we choose SFU for our new project, because we knew its API and flow, and it has very good performance and stability." highly praised Janus. And of course I knew that you have the branch, how could I don't know that? And I even emailed you from the meetecho website to ask for paid service because we do appreciate your work.

But isn't it truth that Janus doesn't have unified plan in your master branch or some stable release? You really want people to use a development branch in a production environment?

Slow is not some bad words, you make things slow and stable. And even @ibc implies Janus is slow `Unified-Plan is not a "super recent and cool WebRTC feature". It's been there for long years.`.

Why are you so mean? I just write my reviews, and I have used and read your code, recommended Janus to people if it fit their needs, I just say that Janus doesn't fit in our specific situation.

**Comment by ibc** [comment]
You have created an audio only SFU for your super specific use case so any other more flexible and powerful SFU is bad for you. Your design comes with tons of bad assumptions just because your single use case. You should know that instead of writing shit about others.

And I never said that Janus is slow just because it doesn't not officially support a specific feature. Your stuff does not support video and you assume "it's an easy extension". You will figure out what slow means by yourself.

**Comment by cedricfung** [comment]
Hi @ibc, you are a good guy, and you have good and delicious soup, and I'm sorry I don't like that taste and I sincerely apologize and you just not suck and bye.

**Comment by cedricfung** [comment]
Since no meaningful things are discussed here, I will lock this thread for good to avoid any further abusive words.

Please only report issues of the software here.
